Add all project files

PavAnufriev opened the pull request.

PavAnufriev committed:
Add all project files

okryukova commented on src/test/java/ru/yandex/praktikum/CreateCourierTest.java:
⛔️Нужно исправить.  Добавь, пожалуйста, пом файл в пулл реквест

okryukova commented on src/test/java/ru/yandex/praktikum/CreateCourierTest.java:
```diff
@@ -0,0 +1,72 @@
+        String firstName = TestDataGenerator.generateRandomString(10);
+
+        Response createResponse = courierClient.createCourier(login, password, firstName);
+        assertEquals(201, createResponse.statusCode());
```
⚠️Можно улучшить. Вместо числовых кодов ответов, лучше использовать коды из библиотеки static org.apache.http.HttpStatus, так как они там строковые и несут смысл, что повышает читаемость и не надо вспоминать, что есть что. Пример кода, statusCode(SC_CREATED).

okryukova commented on src/test/java/ru/yandex/praktikum/CreateCourierTest.java:
```diff
@@ -0,0 +1,72 @@
+        assertEquals(201, createResponse.statusCode());
+        createResponse.then().assertThat().body("ok", equalTo(true));
+
+        Response loginResponse = courierClient.loginCourier(login, password);
```
⛔️Нужно исправить.  Логин нужно перенести в метод с аннотацией after. Шаг должен выполняться, если тест упал

okryukova commented on src/test/java/ru/yandex/praktikum/CreateCourierTest.java:
```diff
@@ -0,0 +1,72 @@
+
+    @Test
+    @DisplayName("Cannot create courier without required fields")
+    @Description("Test verifies that system requires all mandatory fields for courier creation")
```
⛔️Нужно исправить.  Не проверен еще один сценарий : отсутствует логин 
⚠️Можно улучшить. Поле firstName является необязательным для курьера. Попробуй создать курьера без этого поля.

okryukova commented on src/test/java/ru/yandex/praktikum/CreateOrderTest.java:
```diff
@@ -0,0 +1,52 @@
+
+        assertEquals(201, response.statusCode());
+        response.then().assertThat().body("track", notNullValue());
+    }
```
⛔️Нужно исправить. Созданный заказ нужно отменить после выполнения теста

okryukova commented on src/test/java/ru/yandex/praktikum/LoginCourierTest.java:
```diff
@@ -0,0 +1,74 @@
+        
+        assertEquals(404, response.statusCode());
+        response.then().assertThat().body("message", equalTo("Учетная запись не найдена"));
+    }
```
⛔️Нужно исправить. Не проверены еще два кейса:
1. Неверный логин 
2. Пустой логин

okryukova commented on src/test/java/ru/yandex/praktikum/client/CourierClient.java:
```diff
@@ -0,0 +1,39 @@
+        return given()
+                .baseUri(Config.BASE_URL)
+                .header("Content-type", "application/json")
+                .body(String.format("{\"login\": \"%s\", \"password\": \"%s\", \"firstName\": \"%s\"}", 
```
⛔️Нужно исправить. В теле передавать данные нужно через сериализацию. Это более продвинутый вариант, в теории он описан. Поправь, пожалуйста, этот момент во всем коде. Необходимо создать класс CourierModel и описать данные курьера. Затем объект этого класса использовать в тестах и методах

PavAnufriev committed:
Update tests and add POJO models based on review comments:
- Add POJO classes for request/response serialization
- Use HttpStatus constants instead of numeric codes
- Add missing test cases for courier login/creation
- Move login to @After method
- Add order cancellation after tests
- Update API clients to use POJO serialization

okryukova commented on src/test/java/ru/yandex/praktikum/CreateOrderTest.java:
```diff
@@ -0,0 +1,63 @@
+
+        assertEquals(SC_CREATED, response.statusCode());
+        response.then().assertThat().body("track", notNullValue());
+        trackId = response.path("track");
```
⛔️Нужно исправить.  Получение трека нужно вынести в метод tearDown(). Шаг должен выполняться даже если тест упал
